src/boot/efi/efi-string.c

```diff
@@ -272,6 +274,453 @@ bool efi_fnmatch(const char16_t *pattern, const char16_t *haystack) {
 DEFINE_PARSE_NUMBER(char, parse_number8);
 DEFINE_PARSE_NUMBER(char16_t, parse_number16);
 
+enum {
+        SCRATCH_BUF_SIZE = 64,
+};
```

Comment:
btw, I am not sure I agree with your love for enums even for stuff that is not actually enum-like or flags.

The thing is that in particuar for limits you typically want unsigned values actually, but enums are never unsigned. i.e. I think:

```c
#define SCRATCH_BUF_SIZE 64U
```

is much more appropriate actually.

Reply:
Don't bother looking at this code for now. I already majorly reworked this code to make it much nicer to look at. I am just doing some finishing touches right now.